### PR TITLE
Add Common.sln file, refactor solutions to remove unnecessary project references

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

- Created `src/Common/Common.sln` that includes all common projects and gets built before any other solution during build
- Remove all unnecessary project references from solutions so it is able to be built and run all tests on its own

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by maddieclayton** [comment]
sign job here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-sign/815/

File: src/ResourceManager/ApiManagement/Commands.ApiManagement/ChangeLog.md [hunk 1]
```diff
@@ -18,7 +18,11 @@
         - Additional information about change #1
 -->
 ## Current Release
+<<<<<<< HEAD:src/ResourceManager/ApiManagement/ChangeLog.md
```
Comment (maddieclayton) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement/ChangeLog.md]: remove

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/5769/

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/ps-sign/845/

**Pull request merged**